llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -564,6 +561,8 @@ Instruction *InstCombinerImpl::foldSelectIntoOp(SelectInst &SI, Value *TrueVal,
     BinaryOperator *BO =
         BinaryOperator::Create(TVI->getOpcode(), FalseVal, NewSel);
     BO->copyIRFlags(TVI);
+    if (isa<FPMathOperator>(&SI))
+      BO->andIRFlags(NewSel);
```

Comment:
This API is so clumsy.

We could preserve more value flags if the same operands appear in both instructions, but that's probably not worth the complexity